src/main/java/com/example/onlinebookstore/model/ShoppingCart.java

```diff
@@ -0,0 +1,40 @@
+package com.example.onlinebookstore.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "shopping_carts")
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted=false")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne(orphanRemoval = true)
```

Comment:
pls don't leave the default `EAGER` fetch type

also please find out whether `@OneToOne(optional = false` is the same as `@JoinColumn(nullable = false)`

if yes - cool

otherwise - add `optional = false` to `@OneToOne`